File: content/bdd-with-cucumber/10/code/ruby/06-make-buy-test-pass-again/features/step_definitions/shout_steps.rb
When "{person} shouts" do |shouter|
  shout from: shouter, message: "Hello, world"
end

When('{person} shouts {string}') do |shouter, message|
  shout from: shouter, message: message
end

When "{person} shouts the following message" do |shouter, message|
  shout from: shouter, message: message
end

When "{person} shouts {int} over-long messages" do |shouter, count|
  count.times do
    base_message = "A message from #{shouter.name} that is 181 characters long "
    message = base_message + "x" * (181 - base_message.size)
    shout from: shouter, message: message
  end
end

When '{person} shouts {int} messages containing the word {string}' do |shouter, count, word|
  count.times do
    message = "A message containing the word #{word}"
    shout from: shouter, message: message
  end
end

File: content/bdd-with-cucumber/10/code/js/13-make-capitalized-buy-test-pass/features/step_definitions/shout_steps.js
const { When } = require("@cucumber/cucumber")

When("{person} shouts", function (shouter) {
  const message = "Hello, world"
  this.shout({ from: shouter, message })
})

When(
  "{person} shouts {int} messages containing the word {string}",
  function (shouter, count, word) {
    for (let i = 0; i < count; i++) {
      const message = `A message containing the word ${word}`
      this.shout({ from: shouter, message })
    }
  }
)

When("{person} shouts {int} over-long messages", function (shouter, count) {
  for (let i = 0; i < count; i++) {
    const baseMessage = `A message from ${shouter.name} that is 181 characters long `
    const message = baseMessage + "x".repeat(181 - baseMessage.length)
    this.shout({ from: shouter, message })
  }
})

When("{person} shouts {string}", function (shouter, message) {
  this.shout({ from: shouter, message })
})

When("{person} shouts the following message", function (shouter, message) {
  this.shout({ from: shouter, message })
})
